src/main/java/com/amazon/opendistroforelasticsearch/sql/metrics/Metrics.java

```diff
@@ -0,0 +1,88 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.metrics;
+
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class Metrics {
+
+    private static Metrics metrics = new Metrics();
+    private ConcurrentHashMap<String, Metric> registeredMetricsByName = new ConcurrentHashMap<>();
+
+    public static Metrics getInstance() {
+        return metrics;
+    }
+
+    private Metrics() {}
+
+    public void registerDefaultMetrics() {
```

Comment:
how are default metrics different from non-default? If I want to add my own metric, should I add it into the set of default or not?

Reply:
Either is OK, I think. You could add it in this default if you think this metric is should be initialized from beginning. I not, you could choose when and where to register/unregister your own.